### PR TITLE
Edit item implemented.

### PR DESCRIPTION
**Pull request by MertzAndreas**
## Description (READ BEFORE MERGING)

Edit item has been implemented along with additional tests. I have also done some refactoring/name changes. This will still need to be updated with the DatePicker that @Tbjerregaard is working on. This pull requests depends on an update coming from the backend, where I have convinced them to use ISO format in the DTO they send us, meaning that parsing has been simplified on my/our side from now on. I have manually tested the application and have found no problems, but let's all look out for weird behavior.

The backend code will be modified from:
```C#
public static ActivityDTO ToDTO(this Activity activity)
{
    return new ActivityDTO(
        activity.Id,
        activity.CitizenId,
        activity.Date.ToString(),
        activity.Name,
        activity.Description,
        activity.StartTime.ToString("HH:mm"),
        activity.EndTime.ToString("HH:mm"),
        activity.IsCompleted
    );
}
```
To: 
```C#
public static ActivityDTO ToDTO(this Activity activity)
{
    return new ActivityDTO(
        ...
        //This is ISO format which will allow us to directly construct a date object without parsing
        activity.Date.ToString("yyyy-MM-dd"),
        ...
    );
}
```

This Is my PR that tracks the changes: https://github.com/aau-giraf/weekplanner-api/pull/21. Once approved this should be merged (If you don't see any need for changes of course)
## Type of PR

- Feature
- Refactor

## Checklist

- [x] Code follows the code guidelines
- [x] Tests have been added/updated (if applicable)

### CODE REVIEW

File: components/EditTask.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+          display="default"
+          onChange={(_event, selectedDate) => {
+            if (!selectedDate) return;
+            handleInputChange('startTime', selectedDate);
+          }}
+          style={styles.timePicker}
+        />
+      </View>
+
+      <View style={styles.pickerContainer}>
```
Comment (Tbjerregaard) [inline components/EditTask.tsx:121]: Shouldn't we want to ensure endTime is later than startTime somewhere?

**Review by Tbjerregaard — APPROVED** [review]

File: components/EditTask.tsx (hunk 1 above)
Comment (MertzAndreas) [reply components/EditTask.tsx:121]: We should, it's a general theme ive seen but yes we should change it

**Review by MaksimHamko — APPROVED** [review]
I like and I approve =)

File: app/edititem.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+type Params = {
+  name: string;
+  description: string;
+  startTime: string;
+  endTime: string;
+  activityId: string;
+  isCompleted: string;
+};
+
+const HomePage = () => {
```
Comment (Simo205j) [inline app/edititem.tsx:13]: Please change name ❤️

Comment (MertzAndreas) [reply app/edititem.tsx:13]: 🥵

**Review by DanielRBech — APPROVED** [review]
Noice

**Review by Simo205j — APPROVED** [review]

**Review by MaksimHamko — APPROVED** [review]
Nice

**Pull request merged**